File: Code_Chef/DSA learning series/Complexity_Analysis_and_Basic_Warmup/laddu.py
t = int(input())
for i in range(t):
    entrada = input()
    n, local = int(entrada.split()[0]), entrada.split()[1]
    laddus = 0
    for a in range(n):
        entrada = input()
        tipo = entrada.split()[0]
        if tipo == 'CONTEST_WON':
            rank = int(entrada.split()[1])
            laddus += 300 + max(0, 20-rank)
        elif tipo == 'TOP_CONTRIBUTOR':
            laddus += 300
        elif tipo == 'BUG_FOUND':
            severity = int(entrada.split()[1])
            laddus += severity
        elif tipo == 'CONTEST_HOSTED':
            laddus += 50
    if local == 'INDIAN':
        print (laddus // 200)
    elif local == 'NON_INDIAN':
        print (laddus // 400)

File: Code_Chef/DSA learning series/Complexity_Analysis_and_Basic_Warmup/laddu.cpp
#include <iostream>
#include <string>

using namespace std;

int main()
{
    int t, n, rank, severity, pts, ans;
    string origin, activitie;
    
    cin >> t;
    while (t)
    {
        cin >> n;
        cin >> origin;
        pts = 0;
        while (n)
        {
            cin >> activitie;
            if (activitie == "CONTEST_WON")
            {
                cin >> rank;
                pts += 300 + max(0, 20-rank);
            }
            else if (activitie == "TOP_CONTRIBUTOR")
            {
                pts += 300;
            }
            else if (activitie == "BUG_FOUND")
            {
                cin >> severity;
                pts += severity;
            }
            else if (activitie == "CONTEST_HOSTED")
            {
                pts += 50;
            }
            n--;
        }
        if (origin=="INDIAN")
        {
            ans = pts/200;
        }
        else
        {
            ans = pts/400;
        }
        
        cout << ans << endl;
        t--;
    }
    return 0;
}